File: 3931 check-adjacent-digit-differences/solution.java
class Solution {
    public boolean isAdjacentDiffAtMostTwo(String s) {
        int l=1;
        while(l<s.length()) {
            int d1 = s.charAt(l-1) - '0';
            int d2 = s.charAt(l) - '0';
            if(Math.abs(d1-d2) > 2) return false;
            l++;
        }
        return true;
    }
}

File: 3931 check-adjacent-digit-differences/solution.cpp
class Solution {
public:
    bool isAdjacentDiffAtMostTwo(string s) {
        int l=1;
        while(l<s.size()) {
            int d1 = s[l-1] - '0';
            int d2 = s[l] - '0';
            if(abs(d1-d2) > 2) return false;
            l++;
        }
        return true;
    }
};
